mm2src/mm2_bitcoin/spv_validation/src/helpers_validation.rs

```diff
@@ -312,7 +321,63 @@ pub struct BlockHeaderVerificationParams {
     pub difficulty_check: bool,
     pub constant_difficulty: bool,
     pub difficulty_algorithm: Option<DifficultyAlgorithm>,
-    pub genesis_block_header: String,
+    // Where to start fetching block height from. 0 will be the default value if starting_block_header_height is None
+    pub starting_block_header_height: Option<u64>,
+    // Use (starting_block_header_height +1) header hex if starting_block_header_height is set else use genesis block
+    // header hex.
+    pub starting_block_header_hex: String,
+    // retarget_block_header_height is needed if starting_block_header_height is set.
+    pub retarget_block_header_height: Option<u64>,
+    // retarget_block_header_hex is needed if starting_block_header_height is set. (2016)
+    pub retarget_block_header_hex: Option<String>,
```

Comment:
What I meant when we discussed this in DM is that the starting header will have to be a retarget header, you can even rename the fields to `starting_retarget_block_height`, `starting_retarget_block_header_hex`. We also have to ensure that `max_stored_block_headers >= RETARGETING_INTERVAL`

Reply:
ok got it

Reply:
Shouldn't this be `self.max_stored_block_headers.unwrap_or_default() < retarget_interval`
https://github.com/KomodoPlatform/atomicDEX-API/blob/66c228cf01c222f843d619cb5cd390215effa463/mm2src/coins/utxo.rs#L521

Reply:
absolutely thanks 👍🏼